### ISSUE TITLE
Not rendering older areas

### ISSUE THREAD

**Issue by Dolemitez** [overviewer/Minecraft-Overviewer#1570]
Environment:
Java Minecraft 1.14 on Debian Linux.
minecraft-overviewer 0.13.87-0

Problem:
The map was upgraded from 1.13.2.  The only areas rendering are areas visited since the upgrade.  Is the marker that overviewer looks at NULL in un-updated chunks?

After updating the server i used the " --forceUpgrade --eraseCache" switches.

I have run --forcerender and --check-tiles, but still the only areas rendered are areas I've been to since the upgrade as though whatever marker overviewer is checking

**Comment by CounterPillow** [comment]
Can you give me one of the .mca files that doesn't render and throw it up on [the uploader](https://overviewer.org/uploader)? Thanks

**Comment by beeradmoore** [comment]
I was just coming make an issue on this too. There is a Minecraft bug where blocks stop emitting light. You need to call --eraseCache on server launch to fix it. A mod over on the bug report ([MC-142134](https://bugs.mojang.com/browse/MC-142134?focusedCommentId=542933&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-542933)) said 

> Yes, --forceUpgrade upgrades all chunks to 1.14. --eraseCache erases the lighting values saved in the chunks (maybe some other values as well, I don't really know) and re-calculates them as soon as the chunk is loaded the next time by the game. It should be safe to use, but just to be sure, create a backup before.

@CounterPillow would that clearing of light data and now not rendering be related to [this](https://github.com/overviewer/Minecraft-Overviewer/commit/0004540a57677486c46da02eed66f70e3412b7a5) commit?

**Comment by Dolemitez** [comment]
99% certain this is a good example

**Comment by Dolemitez** [comment]
Sorry, didn't notice it didn't attach as an MCA, now zipped

[r.-23.499.mca.zip](https://github.com/overviewer/Minecraft-Overviewer/files/3145088/r.-23.499.mca.zip)

**Comment by CounterPillow** [comment]
Ok cool so these chunks have light data but aren't marked as having light data, I guess we can't rely on anything then.

**Comment by CounterPillow** [comment]
(This is not to be confused with another situation users have run into, where their chunks are marked as having light data, but don't actually have light data)

**Comment by CounterPillow** [comment]
Let me know if this commit makes things better. If yes, I'll close this issue and see it as being fixed.

**Comment by Dolemitez** [comment]
How do I test your commit?  Only some familiarity with git.  I used apt-get install minecraft-overviewer to get to the version I'm at.  Should I just go lazy mode and hunt down that world.py and make those 3 lines happen the way you did.  Says "deletion"?

**Comment by CounterPillow** [comment]
The latest built overviewer version from the apt should contain that commit.

Make sure to `--forcerender`.

**Comment by Dolemitez** [comment]
ok, when I had tried to apt-get update earlier, it didn't find anything new for overviewer, but it triggered this time and I was able to then apt-get install minecraft-overviewer, which performed an update to 0.13.88-0.  Running with --forcerender now.  May take a while, it's an old world without a ton of pruning.  Probably the better part of the day.  Update late tonight US or tomorrow morning if I can get to it before work.  Otherwise, tomorrow evening, US.

**Comment by beeradmoore** [comment]
Bug mentioned earlier creates the lighting issue below. 
![59952487_2348289131860032_2326893021888512000_n](https://user-images.githubusercontent.com/904737/57202645-16154080-6feb-11e9-8d2d-fddafbc4e9fb.jpg)

Minecrafts solution is as OP said, ` --forceUpgrade --eraseCache`, which causes chunks not visited to no longer be rendered. MC mod said this is due to lighting data being cleared and re-calculated on next chunk load. I was able to confirm that existing chunk data still exists when I went to it.
